controller/src/test/java/cern/jarrace/controller/rest/controller/AgentContainerControllerTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    public void testDeployWithDifferentRequestTypes() throws Exception {
+        for (HttpMethod httpMethod : HttpMethod.values()) {
+            MockHttpServletRequestBuilder request = request(httpMethod, "/jarrace/container/deploy/SampleDeploy")
+                    .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                    .content("SampleBytes".getBytes());
+            ResultActions perform = mockMvc.perform(request);
+
+            if (HttpMethod.POST.equals(httpMethod) || HttpMethod.OPTIONS.equals(httpMethod) || HttpMethod.TRACE.equals(httpMethod)) {
+                perform.andExpect(status().isOk());
+            } else {
+                perform.andExpect(status().isMethodNotAllowed());
+            }
+        }
+    }
+
+    @Test
+    public void testDeploy() throws Exception {
+        mockMvc.perform(post("/jarrace/container/deploy/SampleDeploy")
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                .content("SampleBytes".getBytes()));
+        Assert.assertEquals(1, agentContainerController.paths.size());
+        File deployFile = new File(agentContainerController.paths.get("SampleDeploy"));
+        Assert.assertTrue(deployFile.exists());
+        Assert.assertTrue(deployFile.isFile());
+        byte[] bytes = Files.readAllBytes(Paths.get(deployFile.getAbsolutePath()));
+        Assert.assertThat("SampleBytes".getBytes(), equalTo(bytes));
+    }
+
+    @Test
+    public void testRegisterServiceWithDifferentRequestTypes() throws Exception {
+        agentContainerController.entrypoints.put("SampleAgentName", new ArrayList<>());
+        for (HttpMethod httpMethod : HttpMethod.values()) {
+            MockHttpServletRequestBuilder request = request(httpMethod, "/jarrace/service/register/SampleAgentName")
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+                    .content("{\"agentName\":\"SampleAgentName\", \"path\":\"SamplePath\"}");
+            ResultActions perform = mockMvc.perform(request);
+
+            if (HttpMethod.POST.equals(httpMethod) || HttpMethod.OPTIONS.equals(httpMethod) || HttpMethod.TRACE.equals(httpMethod)) {
+                perform.andExpect(status().isOk());
+            } else {
+                perform.andExpect(status().isMethodNotAllowed());
+            }
+        }
+    }
+
+    @Test
+    public void testRegisterServiceWrongMediaType() throws Exception {
+        agentContainerController.entrypoints.put("SampleAgentName", new ArrayList<>());
+        MockHttpServletRequestBuilder request = post("/jarrace/service/register/SampleAgentName")
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                .content("{\"agentName\":\"SampleAgentName\", \"path\":\"SamplePath\"}");
+        mockMvc.perform(request)
+                .andExpect(status().isUnsupportedMediaType());
+    }
+
+    @Test
+    public void testRegisterServiceWithWrongContent() throws Throwable {
```

Comment:
split test into 2, extract common method which accept the content as a parameter. You will then reuse it when testing that the presence of a semicolon is not accepted ;)